core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -214,7 +216,13 @@ class BpmBillingProjectLifecycle(
   private def cleanupLandingZone(
     landingZoneId: UUID,
     ctx: RawlsRequestContext
-  ): DeleteAzureLandingZoneResult = Try(workspaceManagerDAO.deleteLandingZone(landingZoneId, ctx)) match {
+  ): Option[DeleteAzureLandingZoneResult] = Try(workspaceManagerDAO.deleteLandingZone(landingZoneId, ctx)) match {
+    // The service returns a 403 instead of a 404 when there's no landing zone present.
+    // It's fine to move on here, because if this was a 403 for permission reasons, we won't be able to delete the billing profile anyway,
+    // and we don't have a valid instance of a user having access to the billing project but not the underlying resources.
+    case Failure(e: ApiException) if e.getCode == HttpStatus.SC_FORBIDDEN || e.getCode == HttpStatus.SC_NOT_FOUND =>
```

Comment:
I've started pushing this sort of logic down into the DAO layer in the similar BPM error handling [code](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala#L170). The motivation is to keep the HTTP level concerns in the DAO (i.e., StatusCodes.NOT_FOUND etc.) and just return Option[foo] and make the upper levels deal with that interface. What do you think about doing that here?

Reply:
I think that is a better method in the long term, with how we're looking to structure our code, and separate our testing concerns.

I'll give it a try, and see what kind of issues it creates in the immediate term.